multidim_threshold/projection.py

```diff
@@ -54,6 +61,21 @@ def generate_proj_vecs(lo, hi, direc=None):
         yield from vecs
         vecs = [ProjVec(r, direc) for r in next_roots(lo, hi, vecs)]
 
+def generate_axes_intersects(lo, hi, member_oracles, *, searches=None):
+    lo, hi = mdt.map_array((lo, hi))
+    direc = mdt.basis_vecs(len(lo))
+    mids = np.array([fn.nth(0, generate_projections(lo, hi, member_oracles, direc=dir)) for dir in direc])
+    return [axes_mid[:,i].min() for i, axes_mid in enumerate(mids)]
+
+def generate_t_proj_vecs(lo, hi, direc=None):
+    lo, hi = mdt.map_array((lo, hi))
+    if direc is None:
+        direc = hi - lo
+
+    vecs = [ProjVec(lo, direc)]
+    while True:
+        yield from vecs
+        vecs = [ProjVec(random_root(lo, hi), direc)]
```

Comment:
Except for this line, this is the same function as `generate_proj_vecs`.
Maybe we should pass an optional `next_roots` function and pass in a random `next_roots` (which ignores the `vecs` argument).